runtime/common/src/lib.rs

```diff
@@ -50,3 +55,42 @@ macro_rules! prod_or_fast {
 		}
 	};
 }
+
+/// Instance definition for council and technical committee
+pub type CouncilInstance = pallet_collective::Instance1;
+pub type TechnicalCommitteeInstance = pallet_collective::Instance2;
+pub type CouncilMembershipInstance = pallet_membership::Instance1;
+pub type TechnicalCommitteeMembershipInstance = pallet_membership::Instance2;
```

Comment:
This is usage really confused me. Are we allowed to use them in this way? I mean "Instance1" "Instance2", I though they are defined after frame_support::construct_runtime! macro.

Reply:
> This is usage really confused me. Are we allowed to use them in this way? I mean "Instance1" "Instance2", I though they are defined after frame_support::construct_runtime! macro.

yea should be possible, see https://github.com/AcalaNetwork/Acala/blob/d6ad56e209b0c4c08736492d3bc8410deb0c6573/runtime/common/src/lib.rs#L204-L213

Reply:
https://github.com/paritytech/substrate/blob/master/frame/support/src/instances.rs
No worry, I find their hardcode...